### PR TITLE
[user] Add update & delete endpoints

### PR DESCRIPTION
**Pull request by jaylees14**
- Add update and delete endpoints to users
- Update endpoint is a `PUT` so requires all parameters to be provided

Setup:
```
$ curl -X GET localhost/user/2
{"error":"user not found with ID 2"}

$ curl -X POST localhost/user -d '{"name":"Jay"}'
{"ID":2,"Name":"Jay"}

$ curl -X GET localhost/user/2
{"ID":2,"Name":"Jay"}
```

Updating:
```
$ curl -X PUT localhost/user/2 -d '{"name":"Lewis"}'
{}

$ curl -X GET localhost/user/2
{"ID":2,"Name":"Lewis"}
```

Deleting:
```
$ curl -X DELETE localhost/user/2
{}

$ curl -X GET localhost/user/2
{"error":"user not found with ID 2"}
```

### CODE REVIEW

File: user/user.go [hunk 1]
```diff
@@ -83,3 +78,60 @@ func userCreateHandler(w http.ResponseWriter, r *http.Request) {
...
+	err = json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	_, err = valid.ValidateStruct(req)
+	if err != nil {
+		errMsg := CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
```
Comment (Lewiky) [inline user/user.go:99]: Can this whole block be abstracted out? Seems like we do it a lot - maybe for another PR though

**Review by Lewiky — APPROVED** [review]
LGTM

File: user/user.go (hunk 1 above)
Comment (c-harding) [reply user/user.go:99]: Is this happening this PR or next?

Comment (jaylees14) [reply user/user.go:99]: Probably the next

**Review by c-harding — APPROVED** [review]

**Pull request merged**